core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -271,6 +274,7 @@ private JarBuildItem buildUberJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
             List<UberJarMergedResourceBuildItem> mergeResources,
             List<UberJarIgnoredResourceBuildItem> ignoredResources,
             MainClassBuildItem mainClassBuildItem,
+            Optional<NativeAccessBuildItem> nativeAccess,
```

Comment:
I think you need to pass a `List` every time you pass this build item. It's going to be produced from multiple locations and might be produced several times.